### PR TITLE
Log attr fix

### PR DESCRIPTION
**Pull request by felixjulianheitmann**
This is my proposal for a fix of the attributes struct.

It enables a constructor for all integral/floating point types that uses std::to_string, because that is what the spec defines std::to_string for.
It uses a trivial conversion for bools.
It converts all string-like types to strings.

It is C++11 compatible

### CODE REVIEW

**Comment by bachittle** [comment]
keys and values also need to be scrambled, but when I tried it was erroring out (because it can't be called in list initalizer)

File: include/spdlog/details/log_attr.h [hunk 1]
```diff
@@ -1,109 +1,42 @@
...
     std::string key;
     std::string value;
 
 public:
-    attr(Key&& k, Value&& v) : key(std::move(k._key)), value(std::move(v._value)) {}
-    attr(Key const& k, Value const& v) : key(k._key), value(v._value) {}
+    template<typename key_t, typename value_t, typename std::enable_if<is_string<key_t>::value, key_t>::type * = nullptr,
+        typename std::enable_if<is_string<value_t>::value, value_t>::type * = nullptr>
+    attr(key_t const &k, value_t const &v)
+        : key(std::string(k))
```
Comment (bachittle) [inline include/spdlog/details/log_attr.h]: need to call scramble here, otherwise newlines show up in test cases

**Review by bachittle — APPROVED** [review]

File: include/spdlog/details/log_attr.h (hunk 1 above)
Comment (bachittle) [reply include/spdlog/details/log_attr.h]: fixed it

**Comment by felixjulianheitmann** [comment]
Looks good and the tests are passing. Just notice that you now have the implicit conversion of `bool` to  `int` for `std::to_string`. This will generate `0`/`1` instead of `true`/`false` in the resulting json. Which is alright but technically incorrect.

**Pull request closed without merging**